server/controllers/events.js

```diff
@@ -1,61 +1,107 @@
-const Event = require("../models/Event");
+const mongoose = require("mongoose");
+const Joi = require("joi");
+const { Event } = require("../models/Event");
+const httpError = require("../utilities/httpError");
 
 module.exports = {
-  //test function
-  ping: (req, res) => {
-    return res.json({ message: "pong" });
-  },
-  create: async (req, res) => {
+  create: async (req, res, next) => {
     try {
       let data = JSON.parse(req.body.data);
-      data.forEach(obj => obj.user = req.user._id);
+      data.forEach(obj => (obj.user = req.user._id));
       await Event.insertMany(data);
       res.json({ message: "Event created!" });
     } catch (err) {
-      console.log(err);
+      next(err);
     }
   },
-  getAll: async (req, res) => {
+  getAll: async (req, res, next) => {
     try {
-      const events = await Event.find().populate('user').exec();
+      // Get an array of events
+      const events = await Event.find()
+        .populate("user", "displayName")
+        .lean()
+        .exec();
+
       // return all events
       res.json(events);
     } catch (err) {
-      console.log(err);
+      next(err);
     }
   },
-  getOne: async (req, res) => {
+  getOne: async (req, res, next) => {
     try {
-      const event = await Event.findById(req.params.id);
+      const { id } = req.params;
+
+      // Check if the ID is valid
+      if (!mongoose.isValidObjectId(id)) {
+        throw httpError(404);
+      }
```

Comment:
This can be abstracted into a middleware that you can use on each of these routes - not a blocking comment, just a nice-to-have. :)

Reply:
Good point! Will do